### PR TITLE
releng - release prep 0.9.25

### LINKED ISSUE #8429: Dependency pinning in 0.9.24 does not seem to be working
### Describe the bug

Installing via poetry does not use pinned versions, whereas installing via pip does.

In a fresh poetry environment, I did the following:

```shell
poetry add c7n=^0.9.24
```

This installed the following dependencies:

```
  • Installing six (1.16.0)
  • Installing jmespath (1.0.1)
  • Installing python-dateutil (2.8.2)
  • Installing urllib3 (1.26.15)
  • Installing botocore (1.29.98)
  • Installing attrs (22.2.0)
  • Installing pyrsistent (0.19.3)
  • Installing s3transfer (0.6.0)
  • Installing zipp (3.15.0)
  • Installing argcomplete (3.0.4)
  • Installing boto3 (1.26.98)
  • Installing docutils (0.18.1)
  • Installing importlib-metadata (5.2.0)
  • Installing jsonschema (4.17.3)
  • Installing pyyaml (6.0)
  • Installing tabulate (0.9.0)
  • Installing c7n (0.9.24)
```

I expect to be able to export from poetry to a requirements file and use that in a fresh virtual environment.

```export
poetry export --without-hashes -o requirements.txt
poetry run python -m venv test-env 
. test-env/bin/activate
pip install -r requirements.txt 
```

However this currently errors out with conflicting dependencies.

### What did you expect to happen?

I expected the virtual environment to be able to install the versions specified in the `requirements.txt` file.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

_No response_

### Policy

_No response_

### Relevant log/traceback output

_No response_

### Extra information or context

_No response_

### LINKED ISSUE #8432: jinja2 dependency error with mailer 0.6.23
### Describe the bug

After installing mailer 0.6.23 with pip, executing mailer returns this error:

`ModuleNotFoundError: No module named 'jinja2'`

The workaround is to install jinja2 before executing mailer, but I thought this dependency would have been take care of during installation of mailer automatically.

### What did you expect to happen?

Mailer to execute without errors.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
c7n==0.9.24
```

### Policy

_No response_

### Relevant log/traceback output

_No response_

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by kapilt**
tldr the last release had some anomalies with the move to poetry-plugin-freeze, so we want to get a new release out the door that rectifies that.

- releng - 0.9.25 release prep - increment version
- releng - 0.9.25 prep - update deps

fixes #8432 
fixes #8429

### CODE REVIEW

**Review by ajkerrigan — APPROVED** [review]
noticed some flapping w/black, updated mailer to fix that. otherwise lgtm :+1:
